src/main/java/spoon/reflect/declaration/CtElement.java

```diff
@@ -421,4 +421,10 @@
 	// overriding the return type
 	@Override
 	ElementSourceFragment getOriginalSourceFragment();
+
+	/**
+	 * Throws an {@link spoon.JLSViolationException} if enabled, logs otherwise.
+	 * @param reason the reason why this element is not compliant with the JLS.
+	 */
+	void throwJLSViolationException(String reason);
```

Comment:
This method is not related to the element's state. I feel we should have it elsewhere.

Proposal: as static method in SpoonException?

Reply:
The problem with moving it away form `CtElement` is that we need the environment or the factory for the check if we need to throw the exception. I'm unsure if passing a factory or environment to the exception creation produces good code.
```java
public void throwJLSViolation(Factory fac, String reason)
```
looks a bit strange to me.

Reply:
I agree with @monperrus, this method should not be on `CtElement`. It's a helper method for Spoon internals, exposing it in the public API is not desirable.

I also don't really like the idea of having a method named `throwX`, that may not actually throw X. A better name would probably be something like `handleJLSViolation`, but I'm not entirely sold on the necessity for the method to begin with.

Reply:
I moved the Exception to `SpoonException` in a static until method. 

> but I'm not entirely sold on the necessity for the method to begin with.

The goal of this method is to not repeat the check if the exception should be logged or thrown at every place we have JLSViolations.